### PR TITLE
Add tables and histogram for RL staleness

### PR DESCRIPTION
**Pull request by tdene**
# What does this PR do ?

:warning: For major changes (either in lines of code or in its impact), please make sure to first share a design doc with the team. If you're unsure what's the best way to do so, contact the @mcore-oncall.

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### CODE REVIEW

File: megatron/rl/rl_utils.py [hunk 1]
```diff
@@ -929,6 +929,7 @@ def prep_wandb_metrics(
                     rollout_kv_last_token_staleness,
                 )),
             ),
+            # NOTE: This table can get very large (one row per token across all rollouts).
```
Comment (yobibyte) [inline megatron/rl/rl_utils.py:932]: Is the table still usable in this case? We have to user test it before merning.

File: megatron/rl/rl_utils.py [hunk 2]
```diff
@@ -1549,14 +1550,6 @@ def prepare_data_for_update(
                 data = TensorDataset(*dataset_tensors)
                 loader = DataLoader(data, batch_size=args.micro_batch_size)
 
-        with nvtx_range("rl/log-wandb-tb", time=True):
```
Comment (yobibyte) [inline megatron/rl/rl_utils.py:1519]: Why did you remove this and who is doing the logging now?

Comment (tdene) [reply megatron/rl/rl_utils.py:1519]: There were two calls out to `maybe_log_training_metrics`. This removed one is inside `prepare_data_for_update` and ONLY runs on collection iterations (relevant when we collect once and train multiple times). The remaining one is inside `get_grpo_data_iterator` and correctly runs on every iteration.

File: megatron/rl/rl_utils.py (hunk 1 above)
Comment (tdene) [reply megatron/rl/rl_utils.py:932]: I can confirm that on all my less-than-5 steps runs, the table is still alive and kicking in their wandb runs.

But I will make a note to verify the table's health once I have a 40-step run.

File: megatron/rl/rl_utils.py (hunk 2 above)
Comment (yobibyte) [reply megatron/rl/rl_utils.py:1519]: That's actually great, I think our logging for lags>0 have been skewed due to this bug. Good catch!

File: megatron/rl/rl_utils.py [hunk 3]
```diff
@@ -884,8 +884,15 @@ def prep_wandb_metrics(
         data=[[np.mean(g), np.std(g)] for g in rewards],
     )
 
-    true_policy_staleness = [current_iteration - s for g in policy_epoch for s in g]
-    true_kv_staleness = [current_iteration - s for g in kv_cache_epoch for s in g]
+    # Per-rollout staleness (oldest token)
+    rollout_policy_staleness = [current_iteration - min(r) for g in policy_epoch for r in g]
+    rollout_kv_staleness = [current_iteration - min(r) for g in kv_cache_epoch for r in g]
+    # Per-rollout staleness (newest token)
+    rollout_policy_last_token_staleness = [current_iteration - max(r) for g in policy_epoch for r in g]
```
Comment (yobibyte) [inline megatron/rl/rl_utils.py]: why do max() and not r[-1]? sounds like linear vs O(1).

Comment (tdene) [reply megatron/rl/rl_utils.py]: Addressed!
